server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -37,13 +37,11 @@
      * default value is {@link Runtime#availableProcessors()} but should be manually controlled if not all processors on the machine are
      * available to Elasticsearch (e.g., because of CPU limits).
      */
-    public static final Setting<Integer> NODE_PROCESSORS_SETTING = Setting.intSetting(
-        "node.processors",
-        Runtime.getRuntime().availableProcessors(),
-        1,
-        Runtime.getRuntime().availableProcessors(),
-        Property.NodeScope
-    );
+    public static final Setting<Integer> NODE_PROCESSORS_SETTING = createNodeProcessorsSetting(Runtime.getRuntime().availableProcessors());
```

Comment:
Can we instead add a new unregistered setting with the available processors? And use that as the default value for the `node.processors` value. Then the validation could simply override the new setting in the `Settings` object.

Reply:
There's a problem with that approach, if the setting is invalid it will report that the "hidden" setting is invalid and that might be confusing.

Reply:
I'd consider that a bug, since we should validate the input we get to desired nodes - and that should then also be valid on the hidden setting. Perhaps we can validate each internal setting in isolation and report an API level error on the invalid `desired_nodes` parameter (like `processors`)?

Reply:
I'm not sure I understand what you mean. Currently we validate the `node.processors` setting using the provided desired node `processors`. Are you suggesting that we shouldn't override this setting and instead validate it without using the regular settings validation path?

Reply:
We spoke about this offline. I am ok to defer this to a follow-up, the situations where this matters is not that important, but we should address it eventually. We can ensure the validation works by validating the `UpdateDesiredNodesRequest` to ensure it does not contain any invalid specifications (`processors<=0` for instance).